WcaOnRails/app/models/competition_event.rb

```diff
@@ -35,7 +35,7 @@ def event
   def to_wcif
     {
       "id" => self.event.id,
-      "rounds" => self.rounds.order(:number).map(&:to_wcif),
+      "rounds" => self.rounds.map(&:to_wcif),
```

Comment:
rounds are already ordered by number in the associations, and here too there is a hidden N+1 query (one select issued for each round event...) that goes away when using the initial order.